spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -152,8 +152,13 @@ class DeltaCatalog extends DelegatingCatalogExtension
       .getOrElse(spark.sessionState.catalog.defaultTablePath(id))
     val storage = DataSource.buildStorageFormatFromOptions(writeOptions)
       .copy(locationUri = Option(loc))
-    val tableType =
-      if (location.isDefined) CatalogTableType.EXTERNAL else CatalogTableType.MANAGED
+    val isManagedLocation = Option(allTableProperties.get(TableCatalog.PROP_IS_MANAGED_LOCATION))
+      .exists(_.equalsIgnoreCase("true"))
+    val tableType = if (location.isEmpty || isManagedLocation) {
```

Comment:
What will be the impact of this PR with non-UC catalogs like HMS if we backport this to branch 3.2?

Reply:
or should we make it safer to backport by using the [isUnityCatalog](https://github.com/delta-io/delta/blob/89c57836e503a957aae0a5ee073cde44219a7491/spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala#L74) flag here as well?

I think its better we use the flag.